activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -1138,6 +1158,31 @@ def preprocess_order_args(order_args)
         end.flatten!
       end
 
+      # If guard_unsafe_raw_sql is enabled, we only allow column names and
+      # directions as arguments to #order and #reorder. Other arguments will
+      # cause an ArugmentError to be raised.
+      def restrict_order_args(args)
+        return unless ActiveRecord::Base.guard_unsafe_raw_sql
```

Comment:
Would expect the caller to be making the decision whether to check order columns rather than having the check-order method do it as a guard.